src/acl.c

```diff
@@ -1208,6 +1223,8 @@ int ACLSetSelector(aclSelector *selector, const char* op, size_t oplen) {
  * ECHILD: Attempt to allow a specific first argument of a subcommand
  */
 int ACLSetUser(user *u, const char *op, ssize_t oplen) {
+    u->acl_string_dirty = true;
```

Comment:
```suggestion
    u->acl_string_dirty = true;
    if (dst->acl_string) sdsfree(dst->acl_string);
    dst->acl_string = NULL;
```

Reply:
actually just get rid of dirty and just free the string.